core/arch/aarch64/proc.c

```diff
@@ -111,7 +111,13 @@ read_feature_regs(uint64 isa_features[])
         :
         : "x0");
 
-    if (IF_LINUX_ELSE(!IS_STRING_OPTION_EMPTY(xarch_root), false)) {
+    bool skip_aa64mmfr2 = IF_LINUX_ELSE(!IS_STRING_OPTION_EMPTY(xarch_root), false);
+#    ifdef STANDALONE_UNIT_TEST
+    // The core unit_tests call proc_init() and do not have xarch_root set.
```

Comment:
Can't we actually just setenv -xarch_root in DYNAMORIO_OPTIONS in unit_tests? Could do it based on some command line arg passed to unit_tests that tells that it's running under qemu.

Reply:
Or atleast an XXX here.

Reply:
Hmm, that still requires special code, just somewhere else: your logic is that any other xarch_root checks we added in the future in standalone_init code would then also apply to unit_tests?  OK.  Done by adding an env var to the test.

Reply:
> that still requires special code, just somewhere else

When possible, I think it's better to avoid special casing code too much for tests, especially when the test setup can be improved to fall into already handled cases.